src/dataframe/dataframe.jl

```diff
@@ -721,11 +721,25 @@ Base.hcat(df1::DataFrame, df2::AbstractDataFrame, dfn::AbstractDataFrame...) = h
 ## Missing values support
 ##
 ##############################################################################
+"""
+    allowmissing!(df::DataFrame, col::ColumnIndex)
 
+Convert a single column of a DataFrame from element-type `T` to
+`Union{T, Missing}` to support missing values.
+"""
 function allowmissing!(df::DataFrame, col::ColumnIndex)
-    df[col] = Vector{Union{eltype(df[col]), Missing}}(df[col])
+    #TODO use allowmissing from Missings or Base here when possible
+    df[col] = convert(AbstractArray{Union{eltype(df[col]), Missing}}, df[col])
     df
 end
+"""
+    allowmissing!(df::DataFrame[, cols::AbstractVector{<: ColumnIndex}])
+
+Convert columns of a DataFrame in-place from element-type `T` to
+`Union{T, Missing}`. Defaults to converting all columns of the DataFrame,
+although users can manually specify a subset of columns to convert using the
```

Comment:
I don't really like talking about "users" here, it's weird for the "user" who is reading the docstring. :-)